### DISCUSSION TITLE
Difference of Octane and Worker Mode

### DISCUSSION THREAD

**Discussion by masterbater** [php/frankenphp#990] [Q&A] [answered]
We know that php is shared nothing model, is worker mode shared nothing model? I want to use frankenphp like a shared nothing model

**Comment by dunglas** [comment]
Worker mode is not a "share nothing" model. Octane uses FrankenPHP in worker mode.

If you want to use FrankenPHP with a  "share nothing" model, just don't enable the worker mode and PHP will work as usual.

**Reply by masterbater** [reply]
Will this run in worker mode right?
```
FROM dunglas/frankenphp

COPY . /app
```

But I have another question octane has different style of bootstrapping the framework. If I only use worker mode without octane? Will I prevent some gotchas that I encounter in octane? or fundamentally they are the same?

**Reply by dunglas** [reply, marked as answer]
> Will this run in worker mode right?

No, this will run in "normal" mode.

> or fundamentally they are the same?

Octane uses FrankenPHP's worker mode under the hood. You need Octane to use the worker mode (or you'll end up rebuilding something like Octane).
